### PR TITLE
LPS-167302 | Add Autom. test FDSCustomView#CanBeDeleted

### PR DESCRIPTION
**Pull request by mateusralv**
Forwarded manually to liferay-frontend.

PR original: https://github.com/mateusralv/liferay-portal/pull/21

---
**Test Plan:**

- Given A dataset display instance of a component
- And Given A custom view is created
- When Deleting the custom view
- Then Assert the custom view no longer exists

**JIRA Ticket:**
https://issues.liferay.com/browse/LPS-167302

### CODE REVIEW

**Comment by mateusralv** [comment]
ci:test:frontend:nocompile

**Comment by mateusralv** [comment]
The failures are not related, safe to forward 🎉

**Comment by mateusralv** [comment]
ci:forward

**Comment by mateusralv** [comment]
ci:test:frontend:nocompile

**Comment by mateusralv** [comment]
ci:forward

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 1]
```diff
@@ -46,6 +46,22 @@ definition {
 		}
 	}
 
+	@description = "LPS-159575 Automation Test Creation - Allow users to delete their custom viewsLPS"
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:49]: typo with LPS at the end.

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 2]
```diff
@@ -46,6 +46,22 @@ definition {
...
+			FDSCustomView.createNewView(key_nameView = "New View");
+		}
+
+		task ("Deleting the custom view") {
+			FDSCustomView.deleteView();
+		}
+
+		task ("Assert the custom view no longer exists") {
+			FDSCustomView.verifyCustomView(key_nameCustomView = "New View");
+		}
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:62]: Need Given When Then

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 3]
```diff
@@ -46,6 +46,22 @@ definition {
...
+		task ("A custom view is created") {
+			FDSCustomView.createNewView(key_nameView = "New View");
+		}
+
+		task ("Deleting the custom view") {
+			FDSCustomView.deleteView();
+		}
+
+		task ("Assert the custom view no longer exists") {
+			FDSCustomView.verifyCustomView(key_nameCustomView = "New View");
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:61]: Don't use verify in `Then` step.

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro [hunk 4]
```diff
@@ -43,6 +43,24 @@ definition {
...
 	}
 
+	macro deleteView {
+		Click(locator1 = "FrontendDataSet#SELECT_OPTION_HEADER");
+
+		Click(
+			key_itemName = "Delete View",
+			locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW");
+
+		AssertTextEquals(
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:53]: Do not use assertions in macros that are not meant to be in `Then` step assertions.

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro [hunk 5]
```diff
@@ -43,6 +43,24 @@ definition {
...
+			locator1 = "Modal#BODY",
+			value1 = "Are you sure you want to delete this? It will be deleted immediately.");
+
+		Click(
+			key_text = "Delete",
+			locator1 = "Modal#MODAL_FOOTER_BUTTON");
+
+		Alert.viewSuccessMessage();
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:61]: Generally avoid assertions within macros.

**Comment by john-co** [comment]
cc/ @nkkwon fyi for SACRED model best practice pattern feedback.

**Comment by mateusralv** [comment]
Hi @john-co I sent the fix requests in pr: https://github.com/mateusralv/liferay-portal/pull/22 😃

**Pull request closed without merging**